components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1232,6 +1232,33 @@ where
             "to_peer_id" => msg.get_to_peer().get_id(),
         );
 
+        let msg_type = msg.get_message().get_msg_type();
+        let store_id = self.ctx.store_id();
+
+        if disk::disk_full_precheck(store_id) || disk::is_disk_full() {
```

Comment:
Checking an atomic for every raft message and proposal may bring extra cost. 

Maybe we can reduce the cost by only check the atomic once for each batch of peer, more specifically setting a flag like `poller_ctx.is_disk_full` at `PollHandler::begin` and then checking the `poller_ctx.is_disk_full`  for each raft message and proposal instead.

This may bring little delay to detect disk full but I think the delay will be much shorter than the disk stats monitor interval (1s).

Reply:
> Also, I found this PR didn't cover snapshot generating and receiving, those could also consume disk space.
yes, currently, these are also not limited in this PR: snapshot, runninglog, rocksdb I/O, wal, etc.

Reply:
> Checking an atomic for every raft message and proposal may bring extra cost.
> 
> Maybe we can reduce the cost by only check the atomic once for each batch of peer, more specifically setting a flag like `poller_ctx.is_disk_full` at `PollHandler::begin` and then checking the `poller_ctx.is_disk_full` for each raft message and proposal instead.
> 
> This may bring little delay to detect disk full but I think the delay will be much shorter than the disk stats monitor interval (1s).

The impact of access to atomic variables on performance should not be so big, we can add a performance test observation.

Reply:
Not just performance but also resource cost (like CPU), the impact may not big enough to be noticeable but I think it is an unnecessary cost though.